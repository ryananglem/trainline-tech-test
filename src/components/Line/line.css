.lineContainer {
    display: flex;
    height: 80px;  
}
.lineHighlight {
    font-weight: bold;
    color: black;
}
.lineInfo {
    color: #999;
}
.lineIcon {
    width: 85px;
    text-align: center;
    color: black;
}
