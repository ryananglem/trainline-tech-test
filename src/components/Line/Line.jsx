import React from 'react'
import { Station } from '../Station'
import './line.css'

export const Line = (props) => {
        const stations = props.stations
                                .map((s, i) => (<Station 
                                                    key={i} 
                                                    hasDeparted={s.hasDeparted}
                                                    isTrainHere={s.isTrainHere}
                                                    scheduledAt={s.scheduledAt}
                                                    estimatedAt={s.estimatedAt}
                                                    onTime={s.onTime}
                                                    isOrigin={s.isOrigin}
                                                    isDestination={s.isDestination}                                                            
                                                    station={s.station} />))
        return (
        <div>
            <div className="lineContainer">
                <div className="lineIcon">
                    <i className="material-icons md-48">train</i>
                </div>
                <div>
                    <span className="lineHighlight">{ props.departureStation }</span><br/>
                    to <span className="lineHighlight">{ props.destination} </span><br/>
                    <span className="lineInfo">Operated by { props.operator }</span>
                </div>
            </div>
            <div>
                {stations}
            </div>
        </div>)
    }

