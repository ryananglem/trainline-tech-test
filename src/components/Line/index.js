import React, {Component} from 'react'
import {connect} from 'react-redux'
import * as selectors from '../../selectors'
import { Line } from './Line'

export class LineContainer extends Component {

    render() { return ( <Line {...this.props} />
        )
    }    
}
const mapStateToProps = state => {
    const { data } = state
    const operator = data ? data.operator : ""
    const destination = selectors.getDestinationStation(state)
    const origin = selectors.getOriginStation(state)
    const stations = selectors.getStations(state)
    return {
        stations,
        operator,
        destination,
        origin
    }
}
export default connect(mapStateToProps)(Line)