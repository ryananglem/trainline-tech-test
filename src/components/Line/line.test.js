import React from 'react'

import {Line} from './Line'
import renderer from 'react-test-renderer'
import Enzyme, {shallow, mount, render} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

it("renders correctly", () => {
    
    Enzyme.configure({ adapter: new Adapter() })
    const stations = []
    const component = shallow(<Line   departureStation="Farringdon"
                                    origin="origin"
                                    destination="destination"    
                                    operator="operator"
                                    stations={stations} />)
                    
    expect(component).toMatchSnapshot()
})

