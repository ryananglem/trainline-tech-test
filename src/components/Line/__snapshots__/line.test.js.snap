// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    departureStation="Farringdon"
    destination="destination"
    operator="operator"
    origin="origin"
    stations={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="lineContainer"
>
          <div
                    className="lineIcon"
          >
                    <i
                              className="material-icons md-48"
                    >
                              train
                    </i>
          </div>
          <div>
                    <span
                              className="lineHighlight"
                    >
                              Farringdon
                    </span>
                    <br />
                    to 
                    <span
                              className="lineHighlight"
                    >
                              destination
                               
                    </span>
                    <br />
                    <span
                              className="lineInfo"
                    >
                              Operated by 
                              operator
                    </span>
          </div>
</div>,
        <div>
          
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="lineIcon"
>
              <i
                            className="material-icons md-48"
              >
                            train
              </i>
</div>,
            <div>
              <span
                            className="lineHighlight"
              >
                            Farringdon
              </span>
              <br />
              to 
              <span
                            className="lineHighlight"
              >
                            destination
                             
              </span>
              <br />
              <span
                            className="lineInfo"
              >
                            Operated by 
                            operator
              </span>
</div>,
          ],
          "className": "lineContainer",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <i
                className="material-icons md-48"
>
                train
</i>,
              "className": "lineIcon",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "train",
                "className": "material-icons md-48",
              },
              "ref": null,
              "rendered": "train",
              "type": "i",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="lineHighlight"
>
                  Farringdon
</span>,
                <br />,
                "to ",
                <span
                  className="lineHighlight"
>
                  destination
                   
</span>,
                <br />,
                <span
                  className="lineInfo"
>
                  Operated by 
                  operator
</span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Farringdon",
                  "className": "lineHighlight",
                },
                "ref": null,
                "rendered": "Farringdon",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              "to ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "destination",
                    " ",
                  ],
                  "className": "lineHighlight",
                },
                "ref": null,
                "rendered": Array [
                  "destination",
                  " ",
                ],
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Operated by ",
                    "operator",
                  ],
                  "className": "lineInfo",
                },
                "ref": null,
                "rendered": Array [
                  "Operated by ",
                  "operator",
                ],
                "type": "span",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="lineContainer"
>
            <div
                        className="lineIcon"
            >
                        <i
                                    className="material-icons md-48"
                        >
                                    train
                        </i>
            </div>
            <div>
                        <span
                                    className="lineHighlight"
                        >
                                    Farringdon
                        </span>
                        <br />
                        to 
                        <span
                                    className="lineHighlight"
                        >
                                    destination
                                     
                        </span>
                        <br />
                        <span
                                    className="lineInfo"
                        >
                                    Operated by 
                                    operator
                        </span>
            </div>
</div>,
          <div>
            
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="lineIcon"
>
                <i
                                className="material-icons md-48"
                >
                                train
                </i>
</div>,
              <div>
                <span
                                className="lineHighlight"
                >
                                Farringdon
                </span>
                <br />
                to 
                <span
                                className="lineHighlight"
                >
                                destination
                                 
                </span>
                <br />
                <span
                                className="lineInfo"
                >
                                Operated by 
                                operator
                </span>
</div>,
            ],
            "className": "lineContainer",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <i
                  className="material-icons md-48"
>
                  train
</i>,
                "className": "lineIcon",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "train",
                  "className": "material-icons md-48",
                },
                "ref": null,
                "rendered": "train",
                "type": "i",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="lineHighlight"
>
                    Farringdon
</span>,
                  <br />,
                  "to ",
                  <span
                    className="lineHighlight"
>
                    destination
                     
</span>,
                  <br />,
                  <span
                    className="lineInfo"
>
                    Operated by 
                    operator
</span>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Farringdon",
                    "className": "lineHighlight",
                  },
                  "ref": null,
                  "rendered": "Farringdon",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
                "to ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "destination",
                      " ",
                    ],
                    "className": "lineHighlight",
                  },
                  "ref": null,
                  "rendered": Array [
                    "destination",
                    " ",
                  ],
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Operated by ",
                      "operator",
                    ],
                    "className": "lineInfo",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Operated by ",
                    "operator",
                  ],
                  "type": "span",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
